src/cpp/src/whisper_pipeline.cpp

```diff
@@ -105,7 +105,11 @@ ov::genai::WhisperPipeline::WhisperPipeline(const std::string& model_path,
                                             const std::string& device,
                                             const ov::AnyMap& plugin_config) {
     auto start_time = std::chrono::steady_clock::now();
-    m_impl = std::make_unique<WhisperPipeline::Impl>(model_path, tokenizer, device, plugin_config);
+    if (device == "NPU") {
```

Comment:
can device be `NPU.<device ID>` ? Looks like we need to detect whether device name starts with `NPU`

Reply:
`NPU.<device ID>` such format does make sense for compilation on the machine w/o NPU.

But for runtime, NPU should exists on the machine therefore architecture will be detected automatically.

P.S It's a good point and might be useful in case user want to precompile blobs via GenAI, but this flow isn't in-place yet.